addons/hr_expense/models/hr_expense.py

```diff
@@ -312,3 +304,161 @@ def action_get_attachment_view(self):
+        help="The journal used when the expense is done.")
+    accounting_date = fields.Date(default=fields.Date.context_today, string="Accounting Date")
+    bank_journal_id = fields.Many2one('account.journal', string='Bank Journal', states={'done': [('readonly', True)], 'post': [('readonly', True)]}, default=lambda self: self.env['account.journal'].search([('type', 'in', ['case', 'bank'])], limit=1), help="The payment method used when the expense is paid by the company.")
+    account_move_id = fields.Many2one('account.move', string='Journal Entry', copy=False, track_visibility="onchange")
+
+    @api.multi
+    def check_consistency(self):
+        if any(sheet.employee_id != self[0].employee_id for sheet in self):
+            raise UserError(_("Expenses must belong to the same Employee."))
+
+        expense_lines = self.mapped('expense_line_ids')
+        if expense_lines and any(expense.payment_mode != expense_lines[0].payment_mode for expense in expense_lines):
+            raise UserError(_("Expenses must have been paid by the same entity (Company or employee)"))
+
+    @api.model
+    def create(self, vals):
+        sheet = super(HrExpenseSheet, self).create(vals)
+        self.check_consistency()
+        if vals.get('employee_id'):
+            sheet._add_followers()
+        return sheet
+
+    @api.multi
+    def write(self, vals):
+        res = super(HrExpenseSheet, self).write(vals)
+        self.check_consistency()
+        if vals.get('employee_id'):
+            self._add_followers()
+        return res
+
+    @api.multi
+    def _track_subtype(self, init_values):
+        self.ensure_one()
+        if 'state' in init_values and self.state == 'approve':
+            return 'hr_expense.mt_expense_approved'
+        elif 'state' in init_values and self.state == 'submit':
+            return 'hr_expense.mt_expense_confirmed'
+        elif 'state' in init_values and self.state == 'cancel':
+            return 'hr_expense.mt_expense_refused'
+        return super(HrExpenseSheet, self)._track_subtype(init_values)
+
+    def _add_followers(self):
+        user_ids = []
+        employee = self.employee_id
+        if employee.user_id:
+            user_ids.append(employee.user_id.id)
+        if employee.parent_id:
+            user_ids.append(employee.parent_id.user_id.id)
+        if employee.department_id and employee.department_id.manager_id and employee.parent_id != employee.department_id.manager_id:
+            user_ids.append(employee.department_id.manager_id.user_id.id)
+        self.message_subscribe_users(user_ids=user_ids)
+
+    @api.depends('expense_line_ids')
+    def _compute_amount(self):
+        for sheet in self:
+            total = 0.0
+            for line in sheet.expense_line_ids:
+                total += line.total_amount
+            sheet.total_amount = total
+
+    @api.multi
+    def _compute_payment_mode(self):
```

Comment:
It seemed strange to have a related field to a one2many, but as the values are always the same on the same sheet, why not. The method 'traverse_related' should give the first line.